src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -101,43 +109,93 @@ internal CloudTable GetDiagnosticEventsTable(DateTime? now = null)
             return _diagnosticEventsTable;
         }
 
-        private static string GetCurrentTableName(DateTime now)
+        private static string GetTableName(DateTime date)
         {
-            return $"{TableNamePrefix}{now:yyyyMM}";
+            return $"{TableNamePrefix}{date:yyyyMM}";
         }
 
         protected internal virtual async void OnFlushLogs(object state)
         {
             await FlushLogs();
         }
 
+        private async Task PurgePreviousEventVersions()
+        {
+            _logger.LogTrace("Purging previous diagnostic event versions.");
+
+            try
+            {
+                var tables = await TableStorageHelpers.ListTablesAsync(TableClient, TableNamePrefix);
```

Comment:
Should we materialize this IEnumerable? (calling `ToArray` or `ToList) ? Wondering If another add happens while this gets executed, that ends up throwing.

Reply:
I don't think it would throw? But if doing ToList is safer, we can do that!